model_analyzer/record/metrics_manager.py

```diff
@@ -279,13 +280,14 @@ def _get_measurement_if_config_duplicate(self, run_config):
         results = self._state_manager.get_state_variable(
             'ResultManager.results')
 
-        # check whether perf config string is a key in result dict
-        if model_name not in results:
+        if not results.contains_model(model_name):
             return False
-        if not self._is_config_in_results(
-                run_config.model_config()._model_config, results[model_name]):
+
+        if not results.contains_model_config(model_name, model_config_name):
```

Comment:
I would expect that this function can now handle the case above as well (it returns false if the model name doesn't exist, or if the model name exists but the config name doesn't exist in it).